internal-export-file/export-file-csv/src/export-file-csv.py

```diff
@@ -160,132 +160,186 @@ def _process_message(self, data):
+                        'Unknown object type "'
+                        + final_entity_type
+                        + '", doing nothing...'
+                    ),
                 )
-            elif entity_type == "Stix-Core-Object":
-                self.helper.api.stix_core_object.push_list_export(
-                    entity_type, file_name, csv_data, json.dumps(list_params)
+                entities_list = do_list(
+                    search=list_params["search"],
+                    filters=list_params["filters"],
+                    orderBy=list_params["orderBy"],
+                    orderMode=list_params["orderMode"],
+                    relationship_type=list_params["relationship_type"]
+                    if "relationship_type" in list_params
+                    else None,
+                    elementId=list_params["elementId"]
+                    if "elementId" in list_params
+                    else None,
+                    fromId=list_params["fromId"] if "fromId" in list_params else None,
+                    toId=list_params["toId"] if "toId" in list_params else None,
+                    elementWithTargetTypes=list_params["elementWithTargetTypes"]
+                    if "elementWithTargetTypes" in list_params
+                    else None,
+                    fromTypes=list_params["fromTypes"]
+                    if "fromTypes" in list_params
+                    else None,
+                    toTypes=list_params["toTypes"]
+                    if "toTypes" in list_params
+                    else None,
+                    types=list_params["types"] if "types" in list_params else None,
+                    getAll=True,
                 )
-            else:
-                self.helper.api.stix_domain_object.push_list_export(
-                    entity_type, file_name, csv_data, json.dumps(list_params)
+
+                list_filters = json.dumps(list_params)
+
+            if (
+                "element_id" in data and entity_type == "Report"
+            ):  # treatment of reports in entity>Analysis
+                element_id = data["element_id"]
+                if element_id:  # filtering of the data to keep those in the container
+                    new_entities_list = [
+                        entity
+                        for entity in entities_list
+                        if ("objectsIds" in entity)
+                        and (element_id in entity["objectsIds"])
+                    ]
+                    entities_list = new_entities_list
+
+            if entities_list is not None:
+                csv_data = self.export_dict_list_to_csv(entities_list)
+                self.helper.log_info(
+                    "Uploading: " + entity_type + "/" + export_type + " to " + file_name
                 )
-            self.helper.log_info(
-                "Export done: " + entity_type + "/" + export_type + " to " + file_name
-            )
+                if entity_type == "Stix-Cyber-Observable":
```

Comment:
We compute the entities_list (line 313) because entity_type is equald to "Report" (line 303).

I don't undestand how we can have entity_type == "Stix-Cyber-Observable" or "Stix-Core-Object" if entities_list is not None (line 315).

I miss something ?

Reply:
it is an other if that is outside the 'if entity_type == 'Report', so we can have an entity type that is different from 'Report'